### PR TITLE
Deprecate cache implementations

### PR DESCRIPTION
**Pull request by alcaeus**
This PR starts the sunsetting process for doctrine/cache. See #354 for details.

Once this PR is merged, I'll branch 1.11.x and prepare a second PR to drop all deprecated code in preparation for the 1.11 and 2.0 releases.

### CODE REVIEW

**Comment by beberlei** [comment]
This is all good except `CacheProvider`, `Cache`, `FlushableCache`, `ClearableCache`, `MultiOperationCache` should not be deprecated as all third party packages we suggest as alternatives implement `CacheProvider` and would thereby implement something deprecated with 2.0

**Comment by alcaeus** [comment]
I was torn between deprecating them or leaving them as-is. I'll remove the deprecations, we can always do those in 2.1.0.

**Review by greg0ire — DISMISSED** [review]

**Comment by greg0ire** [comment]
No build… I suppose a merge up is needed.

**Review by greg0ire — DISMISSED** [review]

File: composer.json [hunk 1]
```diff
@@ -33,7 +33,9 @@
         "doctrine/coding-standard": "^6.0"
     },
     "suggest": {
-        "alcaeus/mongo-php-adapter": "Required to use legacy MongoDB driver"
+        "alcaeus/mongo-php-adapter": "Required to use legacy MongoDB driver",
+        "psr/cache": "Common interface for caching libraries",
+        "symfony/cache": "PSR-6 and PSR-16 compatible cache libraries with adapters for doctrine/cache"
```
Comment (beberlei) [inline composer.json:38]:
```suggestion
        "symfony/cache": "Use Symfony\Component\Cache\DoctrineProvider to brdige to PSR-6",
        "cache/psr-6-doctrine-bridge": "Use Cache\Bridge\Doctrine\DoctrineCacheBridge to bridge to PSR-6",
```

Comment (derrabus) [reply composer.json:38]: `brdige` --> `bridge`

**Comment by greg0ire** [comment]
The build has been migrated away from travis, let's trigger the new CI pipeline :slightly_smiling_face:

**Review by greg0ire — APPROVED** [review]

**Pull request merged**